tests/server/cluster/cluster_test.go

```diff
@@ -753,20 +753,19 @@ func TestConcurrentHandleRegion(t *testing.T) {
 		re.NoError(err)
 		peerID, err := id.Alloc()
 		re.NoError(err)
-		regionID, err := id.Alloc()
-		re.NoError(err)
 		peer := &metapb.Peer{Id: peerID, StoreId: store.GetId()}
 		regionReq := &pdpb.RegionHeartbeatRequest{
 			Header: testutil.NewRequestHeader(clusterID),
 			Region: &metapb.Region{
-				Id:    regionID,
+				// mock error msg to trigger stream.Recv()
+				Id:    0,
```

Comment:
Will it skip some processes of handle region heartbeat?

Reply:
yes, but this `regionReq` wants to ensure the client can get the server's response.

- only call `SendScheduleCommand` will send right msg by `oc.hbStreams.SendMsg(region, cmd)`
- `DispatchFromHeartBeat` will not trigger `SendScheduleCommand`

Reply:
I don't remember what is the purpose of this test. Does it aim to test the concurrent problem or race problem?

Reply:
Yes, Seems like testing(including race) for concurrent running `HandleRegionHeartbeat`

Reply:
I think core logic for this test is here 
https://github.com/tikv/pd/pull/7998/files#diff-4f23e8da76548a94e6779db8719386a8eabb85ad56deca5e6271a8e162652baaR789-R818

Reply:
If the heartbeat process exits earlier, will it still take effect?

Reply:
I think this part is just like a `ping` which wants to ensure the connection between the client and server.
So I think the msg of the response is unimportant :)

And I have added some breakpoints to ensure concurrent region heartbeat has been handled.